core/lib/zksync_core/src/api_server/web3/state.rs

```diff
@@ -578,13 +582,29 @@ impl Filters {
 
     /// Retrieves filter from the state.
     pub fn get(&self, index: U256) -> Option<&TypedFilter> {
-        self.state.get(&index)
+        self.state.get(&index).map(|installed_filter| {
```

Comment:
Nit: You can use `?` to short-circuit this logic for readability:

```rust
let installed_filter = self.state.get(&index)?;
// ...
Some(&installed_filter.filter)
```

Also, it shouldn't compile without using `get_mut()` and changing the signature to something like `fn get_and_update(&mut self, index: U256)`. Alternatively (I'd prefer this option), you can move all mutation logic to `update()`.